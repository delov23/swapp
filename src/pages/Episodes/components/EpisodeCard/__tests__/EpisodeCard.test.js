import React from 'react';
import { shallow, mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';

import { EpisodeCard } from '../';

const EXAMPLE_PROPS = {
  episode: {
    image: 'image',
    id: 'episodes.1',
    title: 'A new hope',
    openingCrawl: 'LONG TEXT',
  },
};

describe('<EpisodeCard />', () => {
  it('should match snapshot', () => {
    expect(shallow(<EpisodeCard {...EXAMPLE_PROPS} />)).toMatchSnapshot();
  });

  it('should display data', () => {
    const wrapper = mount(
      <Router>
        <EpisodeCard {...EXAMPLE_PROPS} />
      </Router>,
    );

    expect(wrapper.find('h3')).toHaveText(EXAMPLE_PROPS.episode.title);
  });
});
