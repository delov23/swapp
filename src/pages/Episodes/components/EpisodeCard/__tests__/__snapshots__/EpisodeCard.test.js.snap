// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EpisodeCard /> should match snapshot 1`] = `
<Link
  className="cardLight"
  to="/episodes/episodes.1"
>
  <img
    alt=""
    className="cardImg"
    src="image"
  />
  <h3
    className="headingLight"
  >
    A new hope
  </h3>
  <p
    className="paragraph"
  >
    LONG TEXT
  </p>
</Link>
`;
